src/lib/auth/strategies/wpcom/proxy-iframe.js

```diff
@@ -0,0 +1,132 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import { WPcomStrategy } from './index';
+
+/**
+ * Module variables
+ */
+const debug = debugFactory( 'happychat-client:auth:wpcom:proxy' );
+
+/**
+ * WPCOM auth strategy by proxy iframe by using `wpcom` object passed from Calypso. It is assuming
+ * that the customer is authenticated and a `_request` method is created which uses the passed
+ * `wpcom` object to call the WPCOM api. Because the WPCOM endpoints' payload may differ from the
+ * other WPCOM strategy we expose here methods for each WPCOM endpoint that will be used.
+ *
+ * Supported WPCOM oAuth flows:
+ *  - 'wpcom-proxy-iframe'   proxy iframe wpcom authentication using passed wpcom object
+ * @extends WPcomStrategy
+ */
+export default class WPcomProxyIframe extends WPcomStrategy {
+	/**
+	 * Pass the authentication type to the parent and save the auth options and access token.
+	 * @param {Object} authentication information about the authentication type and its options
+	 * @param {string} authentication.type authentication type (wpcom-oauth|wpcom-oauth-by-token)
+	 * @param {Object} authentication.options Set of options sent to auth implementations
+	 */
+	constructor( authentication ) {
+		super( authentication.type );
+		this.options = authentication.options;
+	}
+
+	/**
+	 * Method that creates a request handler which will be used when calling endpoints, it will
+	 * overwrite the placeholder `_request`.
+	 *
+	 * @returns {function} authenticated promisfied wrapper for the passed wpcom object
+	 * @private
+	 */
+	_createAuthenticatedRequest() {
+		/**
+		 * It is basically a wrapper that promisifies and adds authentication to the request
+		 * method of passed wpcom object
+		 *
+		 * @param {Object} args request parameters
+		 * @returns {Promise} authenticated promisfied wrapper for the wpcom object request
+		 */
+		return ( args ) => {
+			return new Promise( ( resolve, reject ) => {
+				this.options.proxy.request( args, ( error, response ) => {
```

Comment:
Any chance `proxy` or `request` are undefined?

Reply:
For proxy there is a check in `/src/lib/auth/index.js`, `request` kinda has to be there but I'll add a check.